core/src/apps/management/recovery_device/homescreen.py

```diff
@@ -48,33 +46,77 @@
     in_progress = storage.recovery.is_in_progress()
     word_count = storage.recovery.get_word_count()
     dry_run = storage.recovery.is_dry_run()
+    backup_type = storage.recovery.get_backup_type()
 
     if not in_progress:  # invalid and inconsistent state
```

Comment:
Well, it makes sure the `_continue_recovery_process` is not called in case there is no recovery in progress. This should not happen. Maybe it's too much?

Reply:
`_continue_recovery_progress` is almost-directly called from `recovery_homescreen`, which only boots with `_boot_recovery`, which is only selected [`if recovery.is_in_progress()`](https://github.com/trezor/trezor-firmware/blob/iss442-supershamir-refactor/core/src/main.py#L91)

There should be a rationale for the check in a comment, or the check shouldn't be here. Otherwise it looks out of place; it's not clear from the context that this should be checked, and that this is the place to do it.

Reply:
The idea was not to call `_continue_recovery_progress` from some wrong place. But I agree, removed.